File: backend/app/schemas/news.py
"""资讯 / 公告相关 Schema。"""

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class NewsBase(BaseModel):
    title: str = Field(..., min_length=1, max_length=200)
    content: str = Field(..., min_length=1)
    full_text: str | None = None  # 原文正文（采集抓取，详情页直接阅读）
    cover_url: str | None = None
    source_url: str | None = None  # 原文链接（详情页内嵌展示）
    category: str = "news"  # news / notice / alert
    author: str | None = None
    is_top: bool = False
    is_published: bool = False


class NewsCreate(NewsBase):
    pass


class NewsUpdate(BaseModel):
    title: str | None = None
    content: str | None = None
    full_text: str | None = None
    cover_url: str | None = None
    source_url: str | None = None
    category: str | None = None
    author: str | None = None
    is_top: bool | None = None
    is_published: bool | None = None


class NewsOut(NewsBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    view_count: int = 0
    created_at: datetime
    updated_at: datetime


class NewsListQuery(BaseModel):
    page: int = 1
    page_size: int = 20
    category: str | None = None
    keyword: str | None = None
    published_only: bool = False

File: frontend-user/src/api/news.ts
import http from './http'

// 资讯分类：news 普通资讯 / notice 官方公告 / alert 气象预警（采集自中央气象台）
export type NewsCategory = 'news' | 'notice' | 'alert'

export interface NewsItem {
  id: number
  title: string
  content: string
  cover_url: string | null
  category: NewsCategory
  author: string | null
  is_top: boolean
  is_published: boolean
  view_count: number
  created_at: string
  updated_at: string
}

export interface NewsListResult {
  items: NewsItem[]
  total: number
  page: number
  page_size: number
}

export interface NewsListParams {
  page?: number
  page_size?: number
  category?: NewsCategory | null
  keyword?: string
  published_only?: boolean
}

// 资讯列表（公开，只取已发布）
export async function listNews(params: NewsListParams = {}): Promise<NewsListResult> {
  return http.get('/news', {
    params: {
      page: params.page ?? 1,
      page_size: params.page_size ?? 12,
      category: params.category ?? undefined,
      keyword: params.keyword ?? undefined,
      published_only: params.published_only ?? true,
    },
  })
}

// 资讯详情（公开，阅读量+1）
export async function getNews(id: number): Promise<NewsItem> {
  return http.get(`/news/${id}`)
}
